packages/render/src/use-composer-action.ts

```diff
@@ -0,0 +1,703 @@
+import type { FarcasterSigner } from "./identity/farcaster";
+
+export type { MiniAppMessage, MiniAppResponse };
+
+type FetchComposerActionFunctionArg = {
+  actionState: ComposerActionState;
+  proxyUrl: string;
+  signer: FarcasterSigner | null;
+  url: string;
+};
+
+type FetchComposerActionFunction = (
+  arg: FetchComposerActionFunctionArg
+) => Promise<void>;
+
+export type RegisterMessageListener = (
+  formResponse: ComposerActionFormResponse,
+  messageListener: MiniAppMessageListener
+) => () => void;
+
+type MiniAppMessageListener = (message: MiniAppMessage) => Promise<void>;
+
+export type OnTransactionFunctionResult = {
+  hash: `0x${string}`;
+  address: `0x${string}`;
+};
+
+export type OnTransactionFunction = (arg: {
+  action: EthSendTransactionAction;
+  address: `0x${string}`;
+}) => Promise<OnTransactionFunctionResult | null>;
+
+export type OnSignatureFunctionResult = {
+  hash: `0x${string}`;
+  address: `0x${string}`;
+};
+
+export type OnSignatureFunction = (arg: {
+  action: EthSignTypedDataV4Action;
+  address: `0x${string}`;
+}) => Promise<OnSignatureFunctionResult | null>;
+
+export type OnCreateCastFunction = (arg: {
+  cast: ComposerActionState;
+}) => Promise<void>;
+
+export type ResolveAddressFunction = () => Promise<`0x${string}` | null>;
+
+export type OnPostMessageToTargetFunction = (
+  response: MiniAppResponse,
+  form: ComposerActionFormResponse
+) => unknown;
+
+export type UseComposerActionOptions = {
+  /**
+   * Current action state, value should be memoized. It doesn't cause composer action / mini app to refetch.
+   */
+  actionState: ComposerActionState;
+  /**
+   * URL to composer action / mini app server
+   *
+   * If value changes it will refetch the composer action / mini app
+   */
+  url: string;
+  /**
+   * Signer used to sign the composer action.
+   *
+   * If value changes it will refetch the composer action / mini app
+   */
+  signer: FarcasterSignerState<any>;
+  /**
+   * URL to the action proxy server. If value changes composer action / mini app will be refetched.
+   *
+   * Proxy must handle POST requests.
+   */
+  proxyUrl: string;
+  /**
+   * If enabled if will fetch the composer action / mini app on mount.
+   *
+   * @defaultValue true
+   */
+  enabled?: boolean;
+  /**
+   * Custom fetch function to fetch the composer action / mini app.
+   */
+  fetch?: (url: string, init: RequestInit) => Promise<Response>;
+  /**
+   * Called before onTransaction/onSignature is invoked to obtain an address to use.
+   *
+   * If the function returns null onTransaction/onSignature will not be called.
+   */
+  resolveAddress: ResolveAddressFunction;
+  onError?: (error: Error) => void;
+  onCreateCast: OnCreateCastFunction;
+  onTransaction: OnTransactionFunction;
+  onSignature: OnSignatureFunction;
+  /**
+   * Called when a response to a message is sent to target (e.g. iframe).
+   */
+  onPostResponseToTarget: OnPostMessageToTargetFunction;
```

Comment:
what do you think of calling this `onMessageRespond`?

currently the doc also reads like the function is going to be called as a side effect instead of handling the actual messaging. perhaps it could read like `Called with message response to be posted to child`

Reply:
I changed it to `onMessageRespond` and changed the comment